### DISCUSSION TITLE
Replace webpack with vite

### DISCUSSION THREAD

**Discussion by erezmus** [eclipse-theia/theia#12503] [Ideas]
Webpack is painfully slow, especially compared to more modern tools like vitejs. Would the community be open to replacing Webpack with Vite?

I realise that the codebase makes heavy use of webpack and also learning a new build tool might take time. One thing that might be possible is to be able to choose between webpack and vite?

**Comment by vince-fugnitto** [comment]
If the benefits from switching are substantial (ex: speed) then I think it can be worth it if you'd like to take a look and contribute the feature. It would be a breaking change for the project and downstream applications however which might include their own `webpack` plugins which would no longer work.

As for supporting both as options I think that would add too much unnecessary complexity, unless you can find a way to keep it clean.
